ceremonies/src/lib.rs

```diff
@@ -152,12 +152,12 @@ decl_module! {
                 ok_or("[EncointerCeremonies]: Overflow adding new participant to registry")?;
             if let Some(p) = proof {
                 // we accept proofs from other communities as well. no need to ensure cid
-                ensure!(sender == p.prover_public, "supplied proof is not proving sender");
-                ensure!(p.ceremony_index < cindex, "proof is acausal");
-                ensure!(p.ceremony_index >= cindex-REPUTATION_LIFETIME, "proof is outdated");
+                ensure!(sender == p.prover_public, Error::<T>::ProofSenderMissing);
+                ensure!(p.ceremony_index < cindex, Error::<T>::ProofAcausal);
+                ensure!(p.ceremony_index >= cindex-REPUTATION_LIFETIME, Error::<T>::ProofOutdated);
                 ensure!(Self::participant_reputation(&(p.community_identifier, p.ceremony_index),
                     &p.attendee_public) == Reputation::VerifiedUnlinked,
-                    "former attendance has not been verified or has already been linked to other account");
+                    Error::<T>::AttendanceUsed);
```

Comment:
it has two possible causes with OR
`AttendanceUnverifiedOrUsed`

Reply:
I would be more expressive here: `AttendanceUnverifiedOrAlreadyUsed`